server/src/main/java/org/opensearch/search/fetch/subphase/highlight/HighlightUtils.java

```diff
@@ -58,24 +60,20 @@ private HighlightUtils() {
     /**
      * Load field values for highlighting.
      */
-    public static List<Object> loadFieldValues(MappedFieldType fieldType, FetchSubPhase.HitContext hitContext, boolean forceSource)
-        throws IOException {
-        // percolator needs to always load from source, thus it sets the global force source to true
-        List<Object> textsToHighlight;
+    public static List<Object> loadFieldValues(
+        MappedFieldType fieldType,
+        QueryShardContext context,
+        FetchSubPhase.HitContext hitContext,
+        boolean forceSource
+    ) throws IOException {
         if (forceSource == false && fieldType.isStored()) {
             CustomFieldsVisitor fieldVisitor = new CustomFieldsVisitor(singleton(fieldType.name()), false);
             hitContext.reader().document(hitContext.docId(), fieldVisitor);
-            textsToHighlight = fieldVisitor.fields().get(fieldType.name());
-            if (textsToHighlight == null) {
-                // Can happen if the document doesn't have the field to highlight
-                textsToHighlight = Collections.emptyList();
-            }
-        } else {
-            SourceLookup sourceLookup = hitContext.sourceLookup();
-            textsToHighlight = sourceLookup.extractRawValues(fieldType.name());
+            List<Object> textsToHighlight = fieldVisitor.fields().get(fieldType.name());
+            return Optional.ofNullable(textsToHighlight).orElse(Collections.emptyList());
```

Comment:
The explicit `if (textsToHighlight == null)` wins here: creating new `Optional` just to check and chain is not worth it (imho).